CDCgov/prime-reportstream
# Add mapping for NIST and tests for mappings

This PR adds mapping for NIST compliant messages including tests for the new datatypes.

Test Steps:
1. Review new datatype tests
2. Run integration tests
3. Review where new datatypes are used in the NIST profile and ensure they are mapped

## Linked Issues
- #13022
